Upstream: no

From 87fb290ca9fb7bbe8f47312d671f8aeed6eaa276 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Thu, 2 Sep 2021 22:47:18 +0200
Subject: [PATCH] FindMySQL: Improve for our installation layout

Our mariadb doesn't install a mysqlclient .pc file and it doesn't
install its libraries directly under lib/.

Change-Id: I08f52a6cbc55682eb985615f0fe9f17987a70336
---
 cmake/FindMySQL.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/cmake/FindMySQL.cmake b/cmake/FindMySQL.cmake
index f5cad361d7..d3d7854585 100644
--- a/cmake/FindMySQL.cmake
+++ b/cmake/FindMySQL.cmake
@@ -40,7 +40,7 @@ if(NOT DEFINED MySQL_ROOT)
     find_package(PkgConfig QUIET)
 endif()
 if(PkgConfig_FOUND AND NOT DEFINED MySQL_ROOT)
-    pkg_check_modules(PC_MySQL QUIET "mysqlclient")
+    pkg_check_modules(PC_MySQL QUIET "mysqlclient" "mariadb")
     set(MySQL_include_dir_hints ${PC_MySQL_INCLUDEDIR})
     set(MySQL_library_hints ${PC_MySQL_LIBDIR})
     set(MySQL_library_hints_debug "")
@@ -61,7 +61,8 @@ find_path(MySQL_INCLUDE_DIR
 find_library(MySQL_LIBRARY

              NO_PACKAGE_ROOT_PATH
              NAMES libmysql mysql mysqlclient libmariadb mariadb
-             HINTS ${MySQL_library_hints})
+             HINTS ${MySQL_library_hints}
+             PATH_SUFFIXES mysql)
 
 if(MySQL_library_hints_debug)
     find_library(MySQL_LIBRARY_DEBUG
-- 
2.40.0

